web/src/test/java/com/replica/StreetEdgeExporterTest.java

```diff
@@ -62,34 +60,80 @@ public static void setUp() {
+    }
+
+    @AfterAll
+    public static void cleanUp() {
+        Helper.removeDir(new File(TARGET_DIR));
+        Helper.removeDir(new File(TRANSIT_DATA_DIR));
+    }
+
+    @Test
+    public void testExportEndToEnd() throws IOException {
         cli.run("export", "test-data/beatty-sample-feed-config-car.yml");
         CSVFormat format = StreetEdgeExporter.CSV_FORMAT;
         File expectedOutputLocation = new File(TARGET_DIR + "street_edges.csv");
         CSVParser parser = CSVParser.parse(expectedOutputLocation, StandardCharsets.UTF_8, format);
         List<CSVRecord> records = parser.getRecords();
         assertEquals(769, records.size());
     }
+
+    @Test
+    public void testExportSingleRecord() {
+        GraphHopperServerConfiguration config = new GraphHopperServerConfiguration();
+        config.getGraphHopperConfiguration().
+                putObject("graph.flag_encoders", "car,foot").
+                putObject("datareader.file", "test-data/beatty.osm").
+                putObject("gtfs.file", "test-data/sample-feed.zip").
+                putObject("graph.location", TARGET_DIR).
+                setProfiles(Collections.singletonList(new Profile("foot").setVehicle("foot").setWeighting("fastest")));
+
+        // TODO copied from ExportCommand
+        // Read in pre-built GH graph files from /transit_data/graphhopper
+        final GraphHopperManaged graphHopper =
+                new GraphHopperManaged(config.getGraphHopperConfiguration(), bootstrap.getObjectMapper());
+        GraphHopper configuredGraphHopper = graphHopper.getGraphHopper();
+        configuredGraphHopper.load(configuredGraphHopper.getGraphHopperLocation());
+
+        // Load OSM info needed for export from MapDB database file
+        DB db = DBMaker.newFileDB(new File("transit_data/osm_info.db")).readOnly().make();
+        Map<Long, Map<String, String>> osmIdToLaneTags = db.getHashMap("osmIdToLaneTags");
+        Map<Integer, Long> ghIdToOsmId = db.getHashMap("ghIdToOsmId");
+        Map<Long, List<String>> osmIdToAccessFlags = db.getHashMap("osmIdToAccessFlags");
+        Map<Long, String> osmIdToStreetName = db.getHashMap("osmIdToStreetName");
+        Map<Long, String> osmIdToHighway = db.getHashMap("osmIdToHighway");
+
+        // Copied from writeStreetEdgesCsv
+        StreetEdgeExporter exporter = new StreetEdgeExporter(configuredGraphHopper, osmIdToLaneTags, ghIdToOsmId, osmIdToAccessFlags, osmIdToStreetName, osmIdToHighway);
+        GraphHopperStorage graphHopperStorage = configuredGraphHopper.getGraphHopperStorage();
+        AllEdgesIterator edgeIterator = graphHopperStorage.getAllEdges();
+
+        // Generate the rows for the first item in the edge iterator
+        edgeIterator.next();
+        List<StreetEdgeExportRecord> records = exporter.generateRecords(edgeIterator);
+        // Expect that two items will be generated
+        assertEquals(2, records.size());
+        // They should be each other's reverse edges
```

Comment:
This is a great idea - maybe we could also check that their line geos are true reverses of each other? It's probably more annoying, because you'll have to convert from string -> geo for both to do it, but seems doable